examples/ensemble/plot_forest_importances_faces.py

```diff
@@ -3,46 +3,91 @@
 Pixel importances with a parallel forest of trees
 =================================================
 
-This example shows the use of forests of trees to evaluate the impurity-based
-importance of the pixels in an image classification task (faces).
-The hotter the pixel, the more important.
+This example shows the use of a forest of trees to evaluate
+the impurity based importance of the pixels in an image
+classification task on the faces dataset.
+The hotter the pixel, the more important it is.
 
 The code below also illustrates how the construction and the computation
 of the predictions can be parallelized within multiple jobs.
 """
+# %%
 print(__doc__)
 
-from time import time
 import matplotlib.pyplot as plt
 
+# %%
+# Loading the data and model fitting
+# ----------------------------------
+# We use the faces data from datasets submodules when using impurity-based
+# feature importance. One drawback of this method is that it cannot be
+# applied on a separate test set.
+# on a separate test set but for this example, we are interested
+# in representing the information learned from the full dataset.
+# Also, we'll set the number of cores to use for the tasks.
 from sklearn.datasets import fetch_olivetti_faces
-from sklearn.ensemble import ExtraTreesClassifier
 
-# Number of cores to use to perform parallel fitting of the forest model
-n_jobs = 1
+# %%
+# We select the number of cores to use to perform parallel fitting of
+# the forest model. `-1` means use all available cores.
+n_jobs = -1
 
+# %%
 # Load the faces dataset
 data = fetch_olivetti_faces()
 X, y = data.data, data.target
 
-mask = y < 5  # Limit to 5 classes
+# %%
+# Limit the dataset to 5 classes.
+mask = y < 5
 X = X[mask]
 y = y[mask]
 
-# Build a forest and compute the pixel importances
-print("Fitting ExtraTreesClassifier on faces data with %d cores..." % n_jobs)
-t0 = time()
-forest = ExtraTreesClassifier(n_estimators=1000,
-                              max_features=128,
-                              n_jobs=n_jobs,
-                              random_state=0)
+# %%
+# A random forest classifier will be fitted to compute the feature importances.
+from sklearn.ensemble import RandomForestClassifier
+
+forest = RandomForestClassifier(
+    n_estimators=750, n_jobs=n_jobs, random_state=42)
 
 forest.fit(X, y)
-print("done in %0.3fs" % (time() - t0))
+
+# %%
+# Feature importance based on mean decrease in impurity (MDI)
+# -----------------------------------------------------------
+# Feature importances are provided by the fitted attribute
+# `feature_importances_` and they are computed as the mean and standard
+# deviation of accumulation of the impurity decrease within each tree.
+#
+# .. warning::
+#     Impurity-based feature importances can be misleading for high cardinality
+#     features (many unique values). See :ref:`permutation_importance` as
+#     an alternative.
+import time
```

Comment:
Moving the `plt` import here makes it easier to tell where the import is from when reading this code block.

```suggestion
import time
import matplotlib.pyplot as plt
```